src/components/ThreadEnvelope.vue

```diff
@@ -212,7 +219,11 @@ export default {
 				logger.debug(`message ${this.envelope.databaseId} fetched`, { message })
 
 				if (!this.envelope.flags.seen) {
-					this.$store.dispatch('toggleEnvelopeSeen', { envelope: this.envelope })
+					logger.info('Starting timer to mark message as seen/read')
+					this.seenTimer = setTimeout(() => {
+						this.$store.dispatch('toggleEnvelopeSeen', { envelope: this.envelope })
+						this.seenTimer = undefined
```

Comment:
I don't think this is necessary

Reply:
what? to keep track of the timerId?

This is needed, I've test it.

I believe the timer is attached to the window, so it's not destroyed when one navigates away from the message

Reply:
No I mean just the part of unsetting the value. I don't think calling clearTimer on a timer ID that has already fired will do any harm. But leave it as is, it's just a nitpick :)